### ISSUE TITLE
Synapse doesn't append `redacts` to the top level for backwards compatibility in v11/MSC2174 rooms

### ISSUE THREAD

**Issue by turt2live** [matrix-org/synapse#15823]
### Description

https://github.com/matrix-org/matrix-spec-proposals/pull/2174 describes:

> For backwards-compatibility with *older* clients, servers should add a `redacts` property to the top level of `m.room.redaction` events in *newer* room versions when serving such events over the Client-Server API.

Empirically, Synapse is not doing this:

![image](https://github.com/matrix-org/synapse/assets/1190097/3eb975b9-c6af-4276-b750-d96096caae33)

Events sent over `/sync` (and possibly `/messages`) are lacking the top level field.

### Steps to reproduce

Create an MSC2174/MSC3820 room and redact a message.

### Homeserver

t2l.io

### Synapse Version

N/A

### Installation Method

pip (from PyPI)

### Database

PostgreSQL

### Workers

Multiple workers

### Platform

N/A

### Relevant log output

```shell
N/A
```

**Comment by clokep** [comment]
I confirmed this is fixed by #15912:

```json
{
  "type": "m.room.redaction",
  "sender": "@alice:localhost:8480",
  "content": {
    "reason": "asdf",
    "redacts": "$w0hBk82GTyF9bRg7w1HLt4iYYP41VYtuhq0UezOol0k"
  },
  "origin_server_ts": 1689683815471,
  "unsigned": {
    "age": 260863,
    "transaction_id": "m1689683815400.5"
  },
  "event_id": "$uF2gT5i6B03SdAXYlcg4FtMzDvxBRV0fRNyC2XKwYQk",
  "redacts": "$w0hBk82GTyF9bRg7w1HLt4iYYP41VYtuhq0UezOol0k"
}
```

**Comment by clokep** [comment]
I realized while writing matrix-org/matrix-spec#1604 that the MSC also says:

> For improved compatibility with *newer* clients, servers should add a `redacts` property to the `content` of `m.room.redaction` events in *older* room versions when serving such events over the Client-Server API.

We don't do this currently and probably should.
